File: app/src/main/java/com/example/angelo/todo/repositories/TaskRepository.java
package com.example.angelo.todo.repositories;

import java.util.ArrayList;

import com.example.angelo.todo.database.TodoDatabaseHelper;
import com.example.angelo.todo.models.Task;

public class TaskRepository {

    TodoDatabaseHelper database;

    public TaskRepository(TodoDatabaseHelper database) {
        this.database = database;
    }

    public long insert(Task task) {
        return this.database.insertTask(task);
    }

    public void delete(Task task) {
        this.database.deleteTask(task);
    }

    public void update(Task task) {
        this.database.updateTask(task);
    }

    public ArrayList<Task> list() {
        ArrayList<Task> taskList = this.database.listTask();

        return taskList;
    }
}

File: app/src/main/java/com/example/angelo/todo/repositories/TaskRepository.kt
package com.example.angelo.todo.repositories

import java.util.ArrayList

import com.example.angelo.todo.database.DatabaseHelper
import com.example.angelo.todo.mapper.TaskMapper
import com.example.angelo.todo.models.Task

class TaskRepository(private val database: DatabaseHelper) {
    fun insert(task: Task): Task {
        val _id = this.database.insertTask(TaskMapper.toInsert(task))

        task.id = _id.toInt()

        return task
    }

    fun delete(task: Task) {
        this.database.deleteTask(task.id)
    }

    fun update(task: Task): Task {
        this.database.updateTask(task.id, task.name!!)

        return task
    }

    fun list(): ArrayList<Task> {
        val taskDataList = this.database.listTask()

        return TaskMapper.toList(taskDataList)
    }
}
